lib/rules/selector-attribute-name-disallowed-list/index.mjs

```diff
@@ -37,24 +37,26 @@ const rule = (primary) => {
 				return;
 			}
 
-			parseSelector(ruleNode.selector, result, ruleNode)?.walkAttributes((attributeNode) => {
-				const attributeName = attributeNode.qualifiedAttribute;
+			flattenNestedSelectorsForRule(ruleNode, result).forEach(({ selector }) => {
```

Comment:
Do we need to switch to `flattenNestedSelectorsForRule`?

This rule only looks at individual attribute selector parts and doesn't do any analysis of the entire selector. So resolving nested selectors shouldn't be needed.

I think we can use the `getRuleSelector` utility instead.

e.g.

```js
parseSelector(getRuleSelector(ruleNode), result, ruleNode)?.walkAttributes((attributeNode) => {
```

Reply:
Got it, using `getRuleSelector` helps to include comments in the selector. I've made the update.

[refactor to use getRuleSelector](https://github.com/stylelint/stylelint/pull/8037/commits/4d81b4ddea71f73da346ead5115e9d2a7db70b8d)